### DISCUSSION TITLE
Nested @tailwind rules were detected

### DISCUSSION THREAD

**Discussion by yevgeniy-belov** [tailwindlabs/tailwindcss#10198] [Help] [answered]
Im trying to use the new CSS Cascade Layers like this:

`@layer tw-base, tw-components, tw-utilities;`

`@layer tw-base {
     @tailwind base
}`
`@layer tw-components {
     @tailwind components
}`
`@layer tw-utilities {
     @tailwind utilities
}`

and Im getting the warning below:
`[vite:css] Nested @tailwind rules were detected, but are not supported.`

Is there a way to get rid of this warning?

Here is the the reproduction:
repo: https://github.com/yevgeniy-belov/vite-react-tw-nesting
codespaces: https://yevgeniy-belov-reimagined-engine-xrjg6pq6wq2vx96.github.dev/

**Comment by wongjn** [comment]
Yes, unnest the `@tailwind` rules:

```css
@tailwind base;
@tailwind components;
@tailwind utilities;
```

**Reply by yevgeniy-belov** [reply]
Unsetting the nesting doesn't help.
Im trying to reduce the specificity of the "components" rules so, I do need a way to define in to which CSS layer it goes...

**Reply by wongjn** [reply]
Seems like you're trying to do something that Tailwind isn't really built to do – any component class you'd like to override could be done by utility class, either due to source order (assuming the component class is at most (0, 1, 0) specificity) or using the `!important` modifier or have the rules that need to override others to have the same or higher specificity and have them appear after the rules they need to override.

**Comment by adamwathan** [comment, marked as answer]
I think this probably counts as a bug, I’d like to improve our story for native cascade layers soon so will look into this one.

**Reply by AshwinBhatkal** [reply]
Has this been looked into?
I'm doing something like this inside the child app to ensure our parent app css is not altered by the child microfrontend being rendered

.root-class-name {
  @tailwind base;
}

With this, I am getting the same warning as above

**Reply by thomiceli** [reply]
+1 on this feature, would be a great addition if this was supported.

**Reply by Sljcom** [reply]
+1

**Reply by Sljcom** [reply]
We are mixing multiple applications in our system temporarily, so multiple CSS files are being used.

We wish to capsulate TailwindCSS within the one application used in the application so that it does not become globally available.

**Reply by alina-holletzek** [reply]
Tailwinds [selector strategy](https://tailwindcss.com/docs/configuration#selector-strategy) helped me to scope styles of each remote app.
I use a `mount` function to integrate the whole remote into my host application, so I can use the `id` of remotes react root div to configure each remote tailwind config via `important`.

**Reply by clicktodev** [reply]
@alina-holletzek selector strategy doesn't scope base styles
